### PR TITLE
Rewrite text around ICE restart in answer. Fixes #671

### LINKED ISSUE #671: Remote desc handling clarification
BLOCKER: Section 5.9 discusses remote description handling. This means that the first bullet on page 65 talks about remote description handling in the "have-local-offer" state. This means that the second sentence of that bullet talks about remote offer handling when in the "have-local-offer" state. That... doesn't seem right. I'm particularly concerned that the language regarding ICE restart probably belongs somewhere else, since it doesn't appear to belong where it is, but I can't quickly figure out where it's missing from.

### PR DESCRIPTION
**Pull request by ekr**

### CODE REVIEW

File: draft-ietf-rtcweb-jsep.xml [hunk 1]
```diff
@@ -3642,27 +3642,34 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
         processing MUST stop and an error MUST be returned.
         <list style="symbols">
 
-          <t>If the PeerConnection state is "have-local-offer", and the
-          ICE ufrag or password changed from the previous remote
+          <t>If the ICE ufrag or password are unchanged from the previous remote description,
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: I read this a few times, and it's technically correct, but it's kind of hard to parse.

I sort of prefer the original outline from the bug which was easier to follow. Was there a reason you chose this different structure?

Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: The original structure actually didn't cover the "if the offer changed and the other side didn't" case, and the text just had it in leaf nodes. So I decided to put it upfront. Happy to talk about other structure.

Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: OK, I understand the intended structure better now. I suggest moving this after the paragraph below, and simplifying the wording to "If the current local description indicates an ICE restart, and the ICE ufrag or password are unchanged from the previous remote description, generate an error".

Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: Maybe also include a ref to 9.2.1.1, as before.

File: draft-ietf-rtcweb-jsep.xml [hunk 2]
```diff
@@ -3642,27 +3642,34 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+          and the description is of type "answer" or "pranswer", and
+          the current local description is an ICE restart, then
+          generate an error.</t>
+
+          <t>If the ICE ufrag or password changed from the previous remote
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: This text is somewhat confusing in the case where the description is an answer - it's not clear what "an ICE restart is needed" means, if the offer the client sent already asked for an ICE restart.

Maybe just say "If the ICE ufrag or password changed from the previous remote description:", and then include the 9.1.1.1 reference in the 'offer' bullet.

File: draft-ietf-rtcweb-jsep.xml [hunk 3]
```diff
@@ -3642,27 +3642,34 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+          </t>
 
-          <t>If the PeerConnection state is "have-remote-pranswer", and
-          the ICE ufrag or password changed from the previous
-          provisional answer, then signal the ICE agent to discard any
-          previous ICE check list state for the m= section and begin
-          checks as if this were the first answer. However, such an
-          answer MAY only change the ICE ufrag or password if the local
-          offer is starting or restarting ICE for the m= section.</t>
+          <t>If the current local description indicates an ICE restart,
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: Justin, I don't love this phrasing, but it was hard to fit the 9.1.1.1 reference in without.

File: draft-ietf-rtcweb-jsep.xml [hunk 4]
```diff
@@ -3642,19 +3642,13 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
           <xref target="RFC5245" />.
           <list style="symbols">
             <t>
               If the description is of type "offer", note that an ICE
-              restart is needed.
+              restart is needed, as described in
+              Section 9.1.1.1 of <xref target="RFC5245"/>.
             </t>
 
             <t>
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: On line 3658, a typo: It -> If

File: draft-ietf-rtcweb-jsep.xml (hunk 3 above)
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: I think this is OK. Some possible edits:
- minor: change 'described' to 'stipulated' or 'prescribed'.
- moderate: 'If the current local description indicates an ICE restart, and the ICE ufrag or password have not changed from the previous remote description, as prescribed by RFC5245, Section 9.2.1.1, generate an error.'

File: draft-ietf-rtcweb-jsep.xml [hunk 5]
```diff
@@ -3642,19 +3642,13 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
-          description, then an ICE restart is needed, as described in
-          Section 9.1.1.1 of
+          <t>If the ICE ufrag or password changed from the previous remote:
           <xref target="RFC5245" />.
           <list style="symbols">
             <t>
               If the description is of type "offer", note that an ICE
-              restart is needed.
+              restart is needed, as described in
+              Section 9.1.1.1 of <xref target="RFC5245"/>.
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: For consistency, suggest flipping this around to [RFC5245], Section 9.1.1.1.

File: draft-ietf-rtcweb-jsep.xml [hunk 6]
```diff
@@ -3671,6 +3665,12 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
           </list>
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: For consistency, suggest flipping line 3663 around to [RFC5245], Section 9.3.1.1.

File: draft-ietf-rtcweb-jsep.xml (hunk 3 above)
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: Thanks, that helped.

**Comment by ekr** [comment]
I took your text and sharpened it a bit
